lib/http2/connection.c

```diff
@@ -317,8 +317,16 @@ static int handle_incoming_request(h2o_http2_conn_t *conn, h2o_http2_stream_t *s
 
     header_exists_map = 0;
     if ((ret = h2o_hpack_parse_headers(&stream->req, &conn->_input_header_table, src, len, &header_exists_map,
-                                       &stream->_expected_content_length, &stream->cache_digests, err_desc)) != 0)
+                                       &stream->_expected_content_length, &stream->cache_digests, err_desc)) != 0) {
+        if (ret == H2O_HTTP2_ERROR_INVALID_HEADER_NAME || ret == H2O_HTTP2_ERROR_INVALID_HEADER_VALUE) {
```

Comment:
Do we need to distinguish between `H2O_HTTP2_ERROR_INVALID_HEADER_NAME` and `H2O_HTTP2_ERROR_INVALID_HEADER_VALUE`?

If not, let's merge them into a single error code. The detail can be found in `*err_desc` anyways.